d2renderers/d2svg/d2svg.go

```diff
@@ -494,6 +495,185 @@ func makeLabelMask(labelTL *geo.Point, width, height int, opacity float64) strin
 	)
 }
 
+// Gets a certain line/curve's SVG path string. offsetIdx and pathData provides the points needed
+func getSVGPathString(pathType *string, offsetIdx int, pathData []string) (string, error) {
+	switch *pathType {
+	case "M":
+		return fmt.Sprintf("M %s %s ", pathData[offsetIdx+1], pathData[offsetIdx+2]), nil
+	case "L":
+		return fmt.Sprintf("L %s %s ", pathData[offsetIdx+1], pathData[offsetIdx+2]), nil
+	case "C":
+		return fmt.Sprintf("C %s %s %s %s %s %s ", pathData[offsetIdx+1], pathData[offsetIdx+2], pathData[offsetIdx+3], pathData[offsetIdx+4], pathData[offsetIdx+5], pathData[offsetIdx+6]), nil
+	case "S":
+		return fmt.Sprintf("S %s %s %s %s ", pathData[offsetIdx+1], pathData[offsetIdx+2], pathData[offsetIdx+3], pathData[offsetIdx+4]), nil
+	default:
+		return "", fmt.Errorf("unknown svg path command \"%s\"", pathData[offsetIdx])
+	}
+}
+
+// Gets how much to increment by on an SVG string to get to the next path command
+func getPathStringIncrement(pathType *string) (int, error) {
```

Comment:
why are you taking the pointer?

Reply:
Sorry, I'm used to languages that copy the entire string so I thought it was more efficient.